plugins/hls-explicit-imports-plugin/src/Ide/Plugin/ExplicitImports.hs

```diff
@@ -476,16 +476,20 @@ filterByImport _ _ = Nothing
 
 constructImport :: LImportDecl GhcRn -> (ModuleName, [AvailInfo]) -> LImportDecl GhcRn
 #if MIN_VERSION_ghc(9,5,0)
-constructImport (L lim imd@ImportDecl {ideclName = L _ _, ideclImportList = Just (hiding, L _ names)})
+constructImport (L lim imd@ImportDecl {ideclName = L _ _, ideclQualified = qualified, ideclImportList = Just (hiding, L _ names)})
```

Comment:
I'm confused - it looks like this case only triggers in the case that the original import *had* an import list! Otherwise `ideclImportList` would be `Nothing` and so we would just go into the fallthrough.

The original bug report suggested that we were converting imports without import lists to imports with import lists, but I can't see how that would happen from this logic. Does the test definitely fail without these changes?

Reply:
Because as it stands, it looks to me like it will take something like this:
```
import RefineA as A (x, y)
```
and turn it into
```
import RefineB as A (x)
import RefineC as A (y)
```
which actually seems fine.

Reply:
(It would also be nice to have a tests for the case of qualified-but-with-import-list also, even if it's weird)

Reply:
It's because on line 343 we actually calculate the refined imports from the generated minimal imports list, not the actual imports list. This is all legacy stuff, but I could see why they would want to do that if they assumed people would always want an explicit import list for refined imports (It certainly helps to know why we are importing some module)

Reply:
> Because as it stands, it looks to me like it will take something like this:
> 
> ```
> import RefineA as A (x, y)
> ```
> 
> and turn it into
> 
> ```
> import RefineB as A (x)
> import RefineC as A (y)
> ```
> 
> which actually seems fine.

That also means that in this case we will end up dumping the explicit import list.

Reply:
From what you're saying, it sounds like the minimal imports code action will also put import lists on qualified imports? That _also_ seems questionable to me - in that situation you can see where something is coming from by looking at the qualifier of the usage, e.g.

```
import Foo qualified as F

F.whatever
```
turning it into 
```
import Foo qualified as F (whatever)

F.whatever
```
Seems worse IMO.

I guess an alternative question is: what if we processed the non-minimal imports in refine imports? Then it seems like we would handle the second case I suggested, since it would be looking at the original imports? It seems reasonable to me for these two things to be orthogonal. WDYT?

Reply:
For minimal imports, we specifically filter out qualified imports (line 324 in this PR). I moved some of the logic around, so now refine imports can work on the original import. For some reason, the edge case is still causing problems (There is a test case for it now, and you should be able to see it failing). Not sure why though.